### ISSUE TITLE
Some bugs and suggestions about processing.

### ISSUE THREAD

**Issue by OneEyeMaker** [BluSunrize/ImmersiveEngineering#425]
1) Crusher doesn't stops after finishing work. For example, I toss stack of ore into this machine, Crusher processes it tp two stacks of dust. But after it Crusher continues to work: consume power, produce sound and show animation until any entity appears in it's hopper.
2) Compatibility bug with Nether Ores. Nether Ores allow to redefine output values for its' processing in config file. For example, on my modpack any ore from this mod smelts to 4 standard ores and crushes to 8 dusts. But IE machines use default values for these operations (2 for smelting and 4 for crushing).
3) Please, add config values for maximum energy input ratio. I want to make refinery very expensive. But if I set energy usage much bigger than 250 RF/t, Refinery starts consume energy faster than it gets one.
4) Please, add support for smelting dusts (crystals/flakes/another dust analogues) from another mods in Arc Furnace. 'Multi-thread' smelting is very cool!

Thank you for your work!
And sorry for my English.

**Comment by cobra** [comment]
1) The crusher issues are already fixed in the upcoming 0.6 release.

**Comment by mindforger** [comment]
2) it's not a bug, but maybe an improvement if IE would be able to implement a compat in future
4) Dust smelting was bugged, wait for the next Version
3) wait what? Can you rephrase?

**Comment by OneEyeMaker** [comment]
3) Add config option to set energy input value. Now, if I set energy usage value in config bigger than 250 RF/t, Refinery consumes energy very fast => energy buffer of Refinery doesn't fill.

**Comment by mindforger** [comment]
this scrambles me, you are sure you have enough input capability? i searched the whole code but there is no chance a 32000RF Buffer can run low while having enough input, this would by the way be the limit of RF/t you could probably achive

**Comment by mindforger** [comment]
scratch that, i found the issue!
new EnergyStorage(32000,256,32000); why @BluSunrize ?

**Comment by BluSunrize** [comment]
Good question.
On the nether ores though, I was fairly sure I'd made them matching. Will need to investigate that.
And I'll change the consumption issues.

**Comment by mindforger** [comment]
make em "matching" means programatically? as @OneEyeMaker stated it is adjustable in the config of nether ores, so maybe there is an API, that's why i mentioned maybe a compat

Question @OneEyeMaker: when you change the netherores config, does this automatically change the output for BOTH the EnderIO SAG Mill AND the ThermalExpansion Grinder? I assume there is no config at all just a compat hook from TE to NetherOres (both share the same developer) and EnderIO SAG Mill does not change it's recipes.

If the second though is correct, consider manipulating the outcome yourself by using MineTweaker the same way as you would have to edit the EnderIO Config to change it's outcome or bother NetherOres for an API :P

**Comment by OneEyeMaker** [comment]
@mindforger 
I don't quite understand your question (My English language learning is still in progress). But I can state the following: changing NetherOres config file (/config/powercrystals/netherores/common.cfg) automatically changes amount of output in all IC2, TE and EIO machines.
If it's handled by NetherOres or by IC2, TE and EIO - I can't say.

**Comment by mindforger** [comment]
thank you that's the information i needed :)

**Comment by mindforger** [comment]
seriously, did you really checked that the SAGMill is chaning it's outputs?
https://github.com/SleepyTrousers/EnderIO/issues/182
this issue from EIO tells me it is not! It is only added with default values to EIO configs! If there is a mechanism that EIO uses to determine changed recipes i did not find it.
for IC2 and TE it is very simple, NetherOres is loading AFTER IC2 and TE and then forcefully manipulates their recipes!

**Comment by mindforger** [comment]
hah ... okay nevermind, i was looking at the wrong maintainer, skyboy took over Netherores
https://github.com/skyboy/NetherOres/pull/31
Like i said, Netherores is hooking into other mods, you should ask over there to get Netherores to hook into IE

**Comment by OneEyeMaker** [comment]
https://github.com/SleepyTrousers/EnderIO/pull/2744

**Comment by BluSunrize** [comment]
I can add support for them, just give me some time, jeez.

**Comment by OneEyeMaker** [comment]
THANK YOU VERY MUCH!
Close the issue, when you're done.

**Comment by mindforger** [comment]
@OneEyeMaker yeah i mentioned the same reference earlier, but this change seem not to adapt any changes you do to the netherores config, it's static! 

i do not see how a change to netherore config could possibly change the recipes of enderio there and netherores seem to not have any API

but i can clearly see, how NEtherores is hooking directly into the SAGMill to enforce it's recipes https://github.com/skyboy/NetherOres/pull/31/files 
i am very sure this is an issue that has to be solved at netherore side

**Comment by BluSunrize** [comment]
It's not. The math is simple @mindforger. Compare the normal ore and the nether ore smelting recipes, determine the output modifier, apply to IE recipes.
Reportign this to skyboy wouldn't really do stuff, he's not that available currently.

**Comment by XFactHD** [comment]
@mindforger this PR didn't get pulled because it contains unstable code (that's what skyboy wrote in the conversation).
But you are right that having it implemented in NetherOres directly is the simplest way.
